resources/src/payment_path.rs

```diff
@@ -0,0 +1,164 @@
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use asset::AssetIdentifier;
+use amount::Amount;
+
+/// A path resource contains information about a payment path. A path can be used by code to
+/// populate necessary fields on path payment operation, such as path and sendMax.  The
+/// also describes assets this path hops through.
+#[derive(Debug)]
+pub struct PaymentPath {
+    path: Vec<AssetIdentifier>,
+    source_amount: Amount,
+    destination_amount: Amount,
+    destination_asset: AssetIdentifier,
+    source_asset: AssetIdentifier,
+}
+
+impl PaymentPath {
+    /// Returns a new Payment Path.
+    pub fn new(
+        path: Vec<AssetIdentifier>,
+        source_amount: Amount,
+        destination_amount: Amount,
+        destination_asset: AssetIdentifier,
+        source_asset: AssetIdentifier,
+    ) -> Result<PaymentPath, String> {
+        Ok(PaymentPath {
+            path,
+            source_amount,
+            destination_amount,
+            destination_asset,
+            source_asset,
+        })
+    }
+
+    /// An array of assets that represents the intermediary assets this path hops through
+    pub fn path<'a>(&'a self) -> &'a Vec<AssetIdentifier> {
+        &self.path
+    }
+
+    /// Destination amount
+    pub fn destination_amount<'a>(&'a self) -> &'a Amount {
+        &self.destination_amount
+    }
+
+    /// Source amount
+    pub fn source_amount<'a>(&'a self) -> &'a Amount {
+        &self.source_amount
+    }
+
+    /// Destination asset specified in the search that found this path
+    pub fn destination_asset<'a>(&'a self) -> &'a AssetIdentifier {
+        &self.destination_asset
+    }
+
+    /// Source asset specified in the search that found this path
+    pub fn source_asset<'a>(&'a self) -> &'a AssetIdentifier {
+        &self.source_asset
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct IntermediatePaymentPath {
```

Comment:
Why does this need to be public?

Reply:
left over from when i was thinking about calling new with an IntermediatePaymentPath.  Thanks for catching